### PR TITLE
Feat/193 english

### PR DESCRIPTION
**Pull request by naomi-eliasar**
This PR aims to solve #192 & #193 & #171

It adds:
- Backend to also save an english title & body
- Clickable flag to switch between the language in the article. This only shows if there are is an english language option
- This is also available in the post preview
- A button in the postform to add an english version of the title & body
- Show the amount of likes of a post in the blog roll

### CODE REVIEW

File: web/src/components/Article/components/ArticlePhotoGallery/ArticlePhotoGallery.tsx [hunk 1]
```diff
@@ -40,7 +40,27 @@ const ArticlePhotoGallery = ({ article, displayType }: Props) => {
...
+                    🇬🇧
+                    <span className="user-select-none absolute bottom-full right-0 mb-2 w-32 rounded-md border-2 border-slate-200 bg-white p-2 text-left text-xs text-slate-500 opacity-0 shadow-md transition-opacity group-hover:opacity-100">
+                      Available in English
+                    </span>
+                  </span>
+                )}
+              </div>
+              <div className="mx-auto max-w-screen-xl px-4 py-20 md:py-24 lg:py-56">
+                <PreviewLayout article={article} />
+              </div>
```
Comment (drikusroor) [inline web/src/components/Article/components/ArticlePhotoGallery/ArticlePhotoGallery.tsx:63]: I think I see this code coming back a couple of times in different files. Can't you make a separate component out of it?

File: web/src/components/Article/components/FullLayout/FullLayout.tsx [hunk 2]
```diff
@@ -22,7 +23,8 @@ interface Props {
 const FullLayout = ({ article }: Props) => {
   const { width } = useWindowDimensions()
   const isMobile = width < 428
-  console.log('ismobile', isMobile)
+
+  const [english, setEnglish] = React.useState<boolean>(false)
```
Comment (drikusroor) [inline web/src/components/Article/components/FullLayout/FullLayout.tsx:27]: Would it be an idea to global preference setting that you can configure in your account settings which is then stored into a cookie?

Comment (drikusroor) [inline web/src/components/Article/components/FullLayout/FullLayout.tsx:27]: Can't you make the default value of this state depend on the availability of an english or dutch version of the post?

if dutch && english => default to dutch
if only english => default to english
if only dutch => default to dutch

File: web/src/components/Article/components/FullLayout/FullLayout.tsx [hunk 3]
```diff
@@ -49,13 +51,39 @@ const FullLayout = ({ article }: Props) => {
...
+                  className="hover:bg-slate-200"
+                  onClick={() => {
+                    setEnglish(true)
+                  }}
+                >
+                  <span className="text-lg">🇬🇧</span>
+                </Button>
+              </div>
+            )}
+          </div>
```
Comment (drikusroor) [inline web/src/components/Article/components/FullLayout/FullLayout.tsx:79]: This language toggler could also be a new component

File: web/src/components/Article/components/FullLayout/FullLayout.tsx [hunk 4]
```diff
@@ -119,8 +147,30 @@ const FullLayout = ({ article }: Props) => {
                   className="pb-1 text-slate-500"
                 />
               </div>
-              <PostThumbsCell postId={article.id} />
+              {article.id && <PostThumbsCell postId={article.id} />}
```
Comment (drikusroor) [inline web/src/components/Article/components/FullLayout/FullLayout.tsx:150]: Nice fix, this was giving those 400 errors in the create article preview right?

File: web/src/components/Article/components/FullLayout/FullLayout.tsx [hunk 5]
```diff
@@ -119,8 +147,30 @@ const FullLayout = ({ article }: Props) => {
...
+                  variant="outlined"
+                  className="hover:bg-slate-200"
+                  onClick={() => {
+                    setEnglish(true)
+                  }}
+                >
+                  <span className="text-xl">🇬🇧</span>
+                </Button>
+              </div>
+            )}
```
Comment (drikusroor) [inline web/src/components/Article/components/FullLayout/FullLayout.tsx:173]: Yeah definitely could become a new component

**Review by drikusroor — APPROVED** [review]
LGTM although I would make new components of the language toggle component and think about whether you can make the dutch/english into a preference

File: web/src/components/Article/components/FullLayout/FullLayout.tsx (hunk 4 above)
Comment (naomi-eliasar) [reply web/src/components/Article/components/FullLayout/FullLayout.tsx:150]: Yes it was!

File: web/src/components/Article/components/FullLayout/FullLayout.tsx (hunk 2 above)
Comment (naomi-eliasar) [reply web/src/components/Article/components/FullLayout/FullLayout.tsx:28]: It is now not possible to have only english, the default title & body are required, whilest the specific english is optional

Comment (naomi-eliasar) [reply web/src/components/Article/components/FullLayout/FullLayout.tsx:28]: Yes, that would be possible indeed
But for now i'm going to merge it like this, pack my bags, and than maybe after do that

**Pull request merged**